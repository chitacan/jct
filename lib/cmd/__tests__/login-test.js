jest.mock('ora');
jest.mock('inquirer');
jest.mock('../../conf');

const {Observable} = require('rxjs');
const {run, subscriber} = require('../login');
const nock = require('nock');

describe('login', () => {
  beforeAll(() => {
    String.prototype.__defineGetter__('red', function() {
      return this;
    });

    String.prototype.__defineGetter__('green', function() {
      return this;
    });

    String.prototype.__defineGetter__('grey', function() {
      return this;
    });
  });

  it('run', (done) => {
    nock('http://localhost/rest', {
      reqheaders: {Authorization: () => true},
    })
    .get('/auth/1/session')
    .reply(200, {});

    Observable.of({})
      .let(run)
      .subscribe((d) => {
        done();
      }, (e) => {
        done.fail(e);
      });
  });

  it('subscriber ok', () => {
    console.log = jest.fn();
    subscriber.next({name: 'name'});
    expect(console.log.mock.calls).toMatchSnapshot();
  });

  it('subscriber error', () => {
    console.log = jest.fn();
    subscriber.error({message: 'error'});
    expect(console.log.mock.calls).toMatchSnapshot();
  });

  afterAll(() => {
    console.log.mockClear();
  });
});
