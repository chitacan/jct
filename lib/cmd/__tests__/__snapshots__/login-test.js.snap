// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login subscriber error 1`] = `
Array [
  Array [
    "
  error
",
  ],
]
`;

exports[`login subscriber ok 1`] = `
Array [
  Array [
    "
  logged in as name
",
  ],
]
`;
